app/src/main/java/org/lineageos/selfie/MainActivity.kt

```diff
@@ -392,6 +399,14 @@ class MainActivity : AppCompatActivity() {
         cameraController.cameraSelector = cameraSelector
         cameraController.setEnabledUseCases(cameraUseCases)
 
+        // Set aspect ratio
+        aspectRatio = sharedPreferences.aspectRatio
+        if (oldY == -1f) {
+            oldY = viewFinder.y
+        }
+        viewFinder.layoutParams.height =
```

Comment:
Wait, this just change the view finder height
What about the actual output?

Reply:
CameraController will adapt the resolution/aspect ratio based on the preview view size
In fact with 4:3 I get a 3000x4000 image, with 16:9 a 2250x4000 image and with 1:1 a 3000x3000 image